test/integration/repos_test.rb

```diff
@@ -12,4 +12,103 @@ class ReposTest < ActionDispatch::IntegrationTest
+      benchmark_result_type: benchmark_result_type,
+      benchmark_type: benchmark_type,
+      initiator: commit
+    )
+
+    benchmark_run2 = create(:commit_benchmark_run,
+      benchmark_result_type: benchmark_result_type,
+      benchmark_type: benchmark_type,
+      initiator: later_commit
+    )
+
+    get "/#{org.name}/#{repo.name}/commits.json?result_type=#{benchmark_type.category}",
+      params: { display_count: 2, result_type: benchmark_result_type }
+
+    res = JSON.parse(response.body, symbolize_names: true)
+
+    assert_includes res.keys, :benchmark_name
+    assert_includes res.keys, :charts
+    assert_includes res.keys, :versions
+    assert_equal res[:charts].length, 1
+    assert_includes res[:charts][0].keys, :measurement
+    assert_includes res[:charts][0].keys, :unit
+    assert_equal res[:charts][0][:datasets].length, 1
+  end
+
+  test "JSON generation works when there are two charts" do
+    benchmark_type = create(:benchmark_type)
+    benchmark_result_type = create(:benchmark_result_type)
+    benchmark_result_type2 = create(:benchmark_result_type)
+
+    repo = benchmark_type.repo
+    org = repo.organization
+
+    commit = create(:commit, repo: repo)
+    later_commit = create(:commit, repo: repo, created_at: 1.day.from_now)
+
+    benchmark_run = create(:commit_benchmark_run,
+      benchmark_result_type: benchmark_result_type,
+      benchmark_type: benchmark_type,
+      initiator: commit
+    )
+    benchmark_run2 = create(:commit_benchmark_run,
+      benchmark_result_type: benchmark_result_type,
+      benchmark_type: benchmark_type,
+      initiator: later_commit
+    )
+
+    benchmark_run3 = create(:commit_benchmark_run,
+      benchmark_result_type: benchmark_result_type2,
+      benchmark_type: benchmark_type,
+      initiator: commit
+    )
+    benchmark_run3 = create(:commit_benchmark_run,
+      benchmark_result_type: benchmark_result_type2,
+      benchmark_type: benchmark_type,
+      initiator: later_commit
+    )
+
+    get "/#{org.name}/#{repo.name}/commits.json?result_type=#{benchmark_type.category}",
+      params: { display_count: 2, result_type: benchmark_result_type }
+
+    res = JSON.parse(response.body, symbolize_names: true)
+
+    assert_includes res.keys, :benchmark_name
+    assert_includes res.keys, :charts
+    assert_includes res.keys, :versions
+    assert_equal res[:charts].length, 2
+    assert_includes res[:charts][0].keys, :measurement
+    assert_includes res[:charts][0].keys, :unit
+    assert_equal res[:charts][0][:datasets].length, 1
+  end
+
+  test "#JSON generation works when there are no charts" do
+    benchmark_type = create(:benchmark_type)
+    benchmark_result_type = create(:benchmark_result_type)
+    repo = benchmark_type.repo
+    org = repo.organization
+
+    get "/#{org.name}/#{repo.name}/commits.json?result_type=#{benchmark_type.category}",
+      params: { display_count: 2, result_type: benchmark_result_type }
```

Comment:
You pass `result_type` as a query param (`result_type=#{benchmark_type.category}"`) and as a form param (`result_type: benchmark_result_type`). Which one do you actually want to pass here? Now at least one of them will be ignored.